### PR TITLE
Track onboarding sources using correlation_id parameter

### PR DESCRIPTION
**Pull request by sirireddy12**
---

This checklist is used to make sure that common guidelines for a pull request are followed.

### General Guidelines

- [ ] Have you run `azdev style <YOUR_EXT>` locally? (`pip install azdev` required)
- [ ] Have you run `python scripts/ci/test_index.py -q` locally?

For new extensions:

- [ ] My extension description/summary conforms to the [Extension Summary Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/extensions/extension_summary_guidelines.md).

### About Extension Publish

There is a pipeline to automatically build, upload and publish extension wheels.  
Once your pull request is merged into main branch, a new pull request will be created to update `src/index.json` automatically.  
The precondition is to put your code inside this repository and upgrade the version in the pull request but do not modify `src/index.json`.

### CODE REVIEW

File: src/connectedk8s/azext_connectedk8s/_params.py [hunk 1]
```diff
@@ -36,6 +36,7 @@ def load_arguments(self, _):
         c.argument('cl_oid', options_list=['--custom-locations-oid'], help="OID of 'custom-locations' app")
         c.argument('onboarding_timeout', options_list=['--onboarding-timeout'], arg_group='Timeout', help='Time required (in seconds) for the arc-agent pods to be installed on the kubernetes cluster. Override this value if the hardware/network constraints on your cluster requires more time for installing the arc-agent pods.')
         c.argument('no_wait', options_list=['--no-wait'], arg_group='Timeout', help="Do not wait for the long-running operation to finish.")
+        c.argument('correlation_id', options_list=['--correlation_id'], help='A guid that is used to track the source of cluster onboarding')
```
Comment (zhoxing-ms) [inline src/connectedk8s/azext_connectedk8s/_params.py:39]:
```suggestion
        c.argument('correlation_id', options_list=['--correlation_id'], help='A guid that is used to track the source of cluster onboarding', validator=override_client_request_id_header)
```

**Comment by yonzhan** [comment]
connectedk8s

File: src/connectedk8s/azext_connectedk8s/custom.py [hunk 2]
```diff
@@ -56,7 +56,7 @@
 # pylint: disable=line-too-long
 
 
-def create_connectedk8s(cmd, client, resource_group_name, cluster_name, https_proxy="", http_proxy="", no_proxy="", proxy_cert="", location=None,
+def create_connectedk8s(cmd, client, resource_group_name, cluster_name, correlation_id=None, https_proxy="", http_proxy="", no_proxy="", proxy_cert="", location=None,
```
Comment (wangzelin007) [inline src/connectedk8s/azext_connectedk8s/custom.py:59]: It seems correlation_id not used in this function.
Can you double check?

Comment (sirireddy12) [reply src/connectedk8s/azext_connectedk8s/custom.py:59]: Hi @wangzelin007 , it is not used in the function, but it is needed as a parameter that should be exposed and it is being set/overridden from a validator. If this parameter is removed from this function, it is not being exposed as a parameter to the "connectedk8s connect" command

**Comment by zhoxing-ms** [comment]
Please write the change of this PR into the history notes `HISTORY.rst`

**Comment by sirireddy12** [comment]
> Please write the change of this PR into the history notes `HISTORY.rst`

Done. Thanks!

**Comment by sirireddy12** [comment]
> > Please write the change of this PR into the history notes `HISTORY.rst`
> 
> Done. Thanks!

@zhoxing-ms Please review the PR

**Review by zhoxing-ms — APPROVED** [review]

**Comment by sirireddy12** [comment]
> > > Please write the change of this PR into the history notes `HISTORY.rst`
> > 
> > 
> > Done. Thanks!
> 
> @zhoxing-ms Please review the PR

@wangzelin007 @zhoxing-ms  Thanks for merging it. Can you please release the latest version?

**Pull request merged**